src/point-cloud-layer.ts

```diff
@@ -0,0 +1,219 @@
+            const { data: table } = this.props;
+        
+            const pointVector = getGeometryVector(table, EXTENSION_NAME.POINT);
+            if (pointVector !== null) {
+              return this._renderLayersPoint(pointVector);
+            }
+        
+            const geometryColumn = this.props.getPosition;
+            if (ga.vector.isPointVector(geometryColumn)) {
+              return this._renderLayersPoint(geometryColumn);
+            }
+        
+            throw new Error("geometryColumn not point or multipoint");
+        }
+
+        _renderLayersPoint(
+            geometryColumn: ga.vector.PointVector,
+        ): Layer<{}> | LayersList | null {
+            const { data: table } = this.props;
+            
+            if (this.props._validate) {
+                assert(ga.vector.isPointVector(geometryColumn));
+                validateAccessors(this.props, table);
+            }
+
+            // Exclude manually-set accessors
+            const [accessors, otherProps] = extractAccessorsFromProps(this.props, [
+                "getPosition",
+                "getNormal",
+                "getColors",
+            ]);
+            const tableOffsets = computeChunkOffsets(table.data);
+
+            const layers: PointCloudLayer[] = [];
+            for (
+                let recordBatchIdx = 0;
+                recordBatchIdx < table.batches.length;
+                recordBatchIdx++
+            ) {
+                const geometryData = geometryColumn.data[recordBatchIdx];
+                // is this actually three d coordinates data?
+                const flatCoordsData = ga.child.getPointChild(geometryData);
+                const flatCoordinateArray = flatCoordsData.values;
+
+                const props: PointCloudLayerProps = {
+                // Note: because this is a composite layer and not doing the rendering
+                // itself, we still have to pass in our defaultProps
+                    ...ourDefaultProps,
+                    ...otherProps,
+
+                    // @ts-expect-error used for picking purposes
+                    recordBatchIdx,
+                    tableOffsets,
+
+                    id: `${this.props.id}-geoarrow-pointcloud-${recordBatchIdx}`,
+                    data: {
+                        length: geometryData.length,
+                        attributes: {
+                            getPosition: {
+                                value: flatCoordinateArray,
+                                size: geometryData.type.listSize,
+                            },
+                            getNormal: {
+                                value: flatCoordinateArray,
+                                size: geometryData.type.listSize,
+                            },
+                            getColors: {
+                                value: flatCoordinateArray,
+                                size: geometryData.type.listSize,
+                            },
```

Comment:
We can take these out because they should be added by `assignAccessor` below.

What's happening right now is you're passing in the _coordinate_ points into the value of `getNormal` and `getColors` as well (note, that's a typo, should be `getColor` iirc). Instead we want to assign 

We have special casing for `getPosition` because we _know_ it's always going to be geometry column input (never a scalar value and never a function callback). For other values it can be either a scalar or a function or a column, and `assignAccessor` is designed to handle that.

Reply:
Okay. Understood.